### PR TITLE
[WIP] Arinax md bzoom

### PR DESCRIPTION
**Pull request by danielhomsa**
The files being added are:
- One modification done to the Microdiff HardwareObject class inside the get_motors() method, the modification is regarding the returned dictionary so when there is no MiniKappa nozzle on the MD this is not returned as motor, avoiding a NoneType exception when using a different headtype.
- Arinax directory: Comtains the HardwareObjects for Arinax Microdiffractometers, Bzoom Camera device and zoom, as well as developments made for the Eiger detector, Vortex detector and other HardwareObjects developed during the work for the SSRF.
- SSRF directory: Contains HardwareObjects modified and developments for the SSRF and the X17UM beamline.

### CODE REVIEW

File: mxcubecore/HardwareObjects/Arinax/ArinaxEiger2X.py [hunk 1]
```diff
@@ -0,0 +1,692 @@
...
+    #        "EnableStream",
+    #        "DisableStream",
+    #    )
+
+    def init(self):
+
+        AbstractDetector.init(self)
+        HardwareObject.init(self)
+
+        self.distance_motor_hwobj = self.get_object_by_role("distance_motor")
```
Comment (rhfogh) [inline mxcubecore/HardwareObjects/Arinax/ArinaxEiger2X.py:84]: am pretty sure this should be simply 'self.distance' according to standard usage.

File: mxcubecore/HardwareObjects/Arinax/ArinaxEiger2X.py [hunk 2]
```diff
@@ -0,0 +1,692 @@
...
+        self.buffer_limit = self.get_property("buffer_limit")
+        self.dcu = self.get_property("dcu")
+        self.det_ip_address = self.get_property("det_ip_address") # 10.30.51.152
+        self.api_version = self.get_property("api_version")
+        self.trigger_mode = self.get_property("trigger_mode") # can be inte, ints, exte, exts
+        self.URL = "http://" + self.det_ip_address + "/%s/api/" + self.api_version + "/%s"
+
+        # config needed to be set up for data collection
+        # if values are None, use the one from the system
+        self.col_config = {
```
Comment (rhfogh) [inline mxcubecore/HardwareObjects/Arinax/ArinaxEiger2X.py:97]: Should be in config file, not hardcoded

File: mxcubecore/HardwareObjects/Arinax/VortexFluoDetector.py [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+
+    def get_roi_count(self):
+        return self.get_channel_object("ROI_Sum_Value").get_value()
+
+    #
+    # def start(self):
+    #     # XSPRESS3-EXAMPLE:Acquire
+    #     pass
+
+    def wait_ready(self, timeout=None):
```
Comment (rhfogh) [inline mxcubecore/HardwareObjects/Arinax/VortexFluoDetector.py:84]: Could you not use the superclass wait_ready, waiting for a ready_event?

**Review by rhfogh — CHANGES_REQUESTED** [review]
Lots of stuff here, and I am not the best person for much of it, so only breezed through this time.

A couple of general points:

Looks like several places made calls to the web (REST) interface. You cannot do that from the core, some people use another interface (Qt) and the dependency goes the other way.

Quite a bit of default values being set inside functions, and not only in mock objects.
For maintainability th8is should be set in configuration, or, at worst, in module level or init code. It is too hard work to go looking in individual functions if you ever need to change a value.

**Review by beteva — COMMENTED** [review]
Dear @danielhomsa,
congratulations for this first PR in mxcubecore. I know it is not easy to follow all the contributing guidelines and maybe not everything is in there, so let me make some suggestions.
It is quite a big PR, What would have been nice is to split the PR in several parts, one for each particular institution. Thus it is easier to look at it.
As a general remark, I think that there is a little mismatch between
- already existing Hardware Objects, which you can use as it, no need to re-write them in Arinax (e.g.Aperture), 
- Deprecated inheritance (like Device in SafetyShutter HO) - it has been decided that we only inherit from HardwareObject,  Device or Equipmet are deprecated.
- Mockup files, which would be nice to have in the mockup directory or do already exist (like EnergyMockup)
- run black --safe and follow python3 syntax.

**Comment by marcus-oscarsson** [comment]
Hi Daniel and Arinax :)

Very nice to see this PR :) :+1: . 

I in general agree with @rhfogh and @beteva. Perhaps what we can do is if you have a look at the comments made above and try to sort out the ones that you think are fairly "accessible" then we can have a quick look together over zoom to go over the the remaining questions. 

Let me know what you think ?

Marcus

**Comment by beteva** [comment]
Dear Arinax team,
looking closely to your PR, it seems that you are not using any of the Hardware Objects, which we thought would be a basis for any equipment, controlled by the MD software,  using the exporter protocol.  Is this deliberate, or were they not suited for your project? The ones which are generic are all entitled Exporter* (e.g.ExporterNState) and some, which are single equipment specific are named Microdiff* (e.g.MicrodiffZoom). I quite agree with  @marcus-oscarsson that a short zoom meeting could be a good idea.

**Comment by marcus-oscarsson** [comment]
I'm marking this as WIP for the time being as there are quite a few questions pending. Please let us know if you have any update on this ?

**Comment by blavault** [comment]
Dear reviewers, thanks for the comments. As Daniel Homs is away I have to dig into this.  I agree that a zoom meeting would be useful because I dont grasp all of the suggestions.
First of all, I want to push the configuration associated to the code  (was missing). 
Can I get access rights for user blavault (bernard.lavault@arinax.com) ? My first push to the arinax_md_bzoom branch failed (remote: Permission to mxcube/mxcubecore.git denied to blavault). Who can arrange that ?

Bernard

**Comment by marcus-oscarsson** [comment]
@blavault You should have write permissions now

**Comment by blavault** [comment]
@marcus-oscarsson  Thanks for access rights , I can push now.

**Comment by marcus-oscarsson** [comment]
@blavault, closing this for now, we can reopen it at a later stage. Let me know if you would like to synchronize regarding this :) ?

**Pull request closed without merging**